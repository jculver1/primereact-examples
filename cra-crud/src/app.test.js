import {render, screen} from "@testing-library/react";
import App from './App'
import userEvent from "@testing-library/user-event";

test("Add a new product to the table", async () => {
  
  render(<App/>);
  let populatedData = await screen.findByText(/black watch/i);
  expect(populatedData).toBeTruthy()
  const totalProducts = await screen.findAllByRole('row').length

  const addProductButton = screen.getByRole("button", {
    name: "add-new-product",
  });

    userEvent.click(addProductButton);

    const inputFieldProdName = screen.getByLabelText('Name', {selector: 'input'});
    userEvent.type(inputFieldProdName, "A Test Name");

    const textareaProdDescription =  screen.getByLabelText('Description', {selector: 'textarea'});
    userEvent.type(textareaProdDescription, "A Test Description of the product");

    const selectProdCategory = screen.getByLabelText('Accessories');
    userEvent.click(selectProdCategory);
       
    const saveAddProduct = screen.getByRole("button", {
        name: "save",
      });

    userEvent.click(saveAddProduct);

    const newTotalOfProducts = await screen.findAllByRole('row')
    expect(newTotalOfProducts === totalProducts + 1)
   
});


test("Delete a product from table", async () => {
  render(<App/>);
  let populatedData = await screen.findByText(/black watch/i);
  expect(populatedData).toBeTruthy()

  const products = await screen.findAllByRole('row')
  const numberOfRowsAfterDelete = products.length - 1;
  const deleteButton = await screen.findAllByRole('button', {
    name: /delete/i
  })

  userEvent.click(deleteButton[1])
 
  const confirmDelete = await screen.findByText(/yes/i)
  userEvent.click(confirmDelete)

  expect((await screen.findAllByRole('row')).length === numberOfRowsAfterDelete)

});