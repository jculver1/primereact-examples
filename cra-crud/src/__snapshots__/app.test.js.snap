// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test crud app with snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="datatable-crud-demo surface-card p-4 border-round shadow-2"
      >
        <div
          class="p-toast p-component p-toast-top-right"
        >
          <div />
        </div>
        <div
          class="text-3xl text-800 font-bold mb-4"
        >
          PrimeReact CRUD
        </div>
        <div
          class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll"
          data-scrollselectors=".p-datatable-wrapper"
          pr_id_18=""
        >
          <div
            class="p-datatable-header"
          >
            <div
              class="flex flex-column md:flex-row md:align-items-center justify-content-between"
            >
              <span
                class="p-input-icon-left w-full md:w-auto"
              >
                <i
                  class="pi pi-search"
                />
                <input
                  class="p-inputtext p-component w-full lg:w-auto"
                  placeholder="Search..."
                  type="search"
                />
              </span>
              <div
                class="mt-3 md:mt-0 flex justify-content-end"
              >
                <button
                  aria-label="add-new-product"
                  class="p-button p-component mr-2 p-button-rounded p-button-icon-only"
                >
                  <span
                    class="p-button-icon p-c pi pi-plus"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                     
                  </span>
                </button>
                <button
                  class="p-button p-component p-button-danger mr-2 p-button-rounded p-button-icon-only p-disabled"
                  disabled=""
                >
                  <span
                    class="p-button-icon p-c pi pi-trash"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                     
                  </span>
                </button>
                <button
                  class="p-button p-component p-button-help p-button-rounded p-button-icon-only"
                >
                  <span
                    class="p-button-icon p-c pi pi-upload"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                     
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="p-datatable-wrapper"
          >
            <table
              class="p-datatable-table"
              role="table"
            >
              <thead
                class="p-datatable-thead"
              >
                <tr
                  role="row"
                >
                  <th
                    class="p-selection-column"
                    role="columnheader"
                    style="width: 3rem;"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      />
                      <div
                        class="p-checkbox p-component"
                      >
                        <div
                          class="p-checkbox-box p-component p-disabled"
                          role="checkbox"
                        >
                          <span
                            class="p-checkbox-icon"
                          />
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 12rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Code
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 16rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Name
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    class=""
                    role="columnheader"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Image
                      </span>
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 8rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Price
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 10rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Category
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 12rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Reviews
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    style="min-width: 12rem;"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        Status
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    class=""
                    role="columnheader"
                    style="min-width: 8rem;"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      />
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="p-datatable-tbody"
              >
                <tr
                  class="p-datatable-emptymessage"
                  role="row"
                >
                  <td
                    colspan="9"
                    role="cell"
                  >
                    No available options
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="p-paginator p-component p-paginator-bottom"
          >
            <button
              class="p-paginator-first p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-left"
              />
            </button>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-left"
              />
            </button>
            <span
              class="p-paginator-pages"
            />
            <button
              class="p-paginator-next p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-right"
              />
            </button>
            <button
              class="p-paginator-last p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-right"
              />
            </button>
            <span
              class="p-paginator-current"
            >
              Showing 0 to 0 of 0 products
            </span>
            <div
              class="p-dropdown p-component p-inputwrapper p-disabled p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  disabled=""
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="datatable-crud-demo surface-card p-4 border-round shadow-2"
    >
      <div
        class="p-toast p-component p-toast-top-right"
      >
        <div />
      </div>
      <div
        class="text-3xl text-800 font-bold mb-4"
      >
        PrimeReact CRUD
      </div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_18=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="flex flex-column md:flex-row md:align-items-center justify-content-between"
          >
            <span
              class="p-input-icon-left w-full md:w-auto"
            >
              <i
                class="pi pi-search"
              />
              <input
                class="p-inputtext p-component w-full lg:w-auto"
                placeholder="Search..."
                type="search"
              />
            </span>
            <div
              class="mt-3 md:mt-0 flex justify-content-end"
            >
              <button
                aria-label="add-new-product"
                class="p-button p-component mr-2 p-button-rounded p-button-icon-only"
              >
                <span
                  class="p-button-icon p-c pi pi-plus"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
              <button
                class="p-button p-component p-button-danger mr-2 p-button-rounded p-button-icon-only p-disabled"
                disabled=""
              >
                <span
                  class="p-button-icon p-c pi pi-trash"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
              <button
                class="p-button p-component p-button-help p-button-rounded p-button-icon-only"
              >
                <span
                  class="p-button-icon p-c pi pi-upload"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  class="p-selection-column"
                  role="columnheader"
                  style="width: 3rem;"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    />
                    <div
                      class="p-checkbox p-component"
                    >
                      <div
                        class="p-checkbox-box p-component p-disabled"
                        role="checkbox"
                      >
                        <span
                          class="p-checkbox-icon"
                        />
                      </div>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 12rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Code
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 16rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Image
                    </span>
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 8rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Price
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 10rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Category
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 12rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Reviews
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  style="min-width: 12rem;"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      Status
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                  style="min-width: 8rem;"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="9"
                  role="cell"
                >
                  No available options
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom"
        >
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-left"
            />
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-left"
            />
          </button>
          <span
            class="p-paginator-pages"
          />
          <button
            class="p-paginator-next p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-right"
            />
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-right"
            />
          </button>
          <span
            class="p-paginator-current"
          >
            Showing 0 to 0 of 0 products
          </span>
          <div
            class="p-dropdown p-component p-inputwrapper p-disabled p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                disabled=""
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
